### DISCUSSION TITLE
Kohya is not working for me.

### DISCUSSION THREAD

**Discussion by Skettalee** [bmaltais/kohya_ss#1446] [Q&A]
I have redownloaded Kohya even numerous times and I followed the instructions to install and get it running but I have tried any captioning and also tried to do any training and all the thing comes back with is I assume errors because some of it is red, i cant decipher but maybe someone else could?

I posted the code to pastebin: https://pastebin.com/qcQukfWP

Can anyone tell me what is going wrong?

**Comment by rpfile** [comment]
ValueError: bf16 mixed precision requires PyTorch >= 1.10 and a supported device.

the error says that either an old version of PyTorch was installed (should be PyTorch 2.x.x) or for some reason mixed precision isn't supported on your GPU. you can try turning off mixed precision entirely in the gui.

**Comment by Skettalee** [comment]
Ive asked this question on reddit stable diffusion community and they mostly told me to revert back to Torch 1, so I did that and it didn't work. still. This is the error I get on that version: https://pastebin.com/q5qCNjg6

I used kohya once before if anything just to do the auto captioning feature and it worked before. I had removed it and Ive added it again wanting to do more training. I have an Nvidia RTX 2080 graphics card, it should be supported. And I never went and downloaded torch i just run the kuhya setup and pick from the 2 options of 1 and 2 and it installs it for me.

**Comment by rpfile** [comment]
well you might have torch 2 installed, the error could be that your device does not support mixed precision.

why not just try turning of mixed precision and see what it does.

with respect to torch1, 

that error:

  File "F:\sd.CΩDE\kohya\kohya_ss\finetune\tag_images_by_wd14_tagger.py", line 15, in <module>
    import library.train_util as train_util
ModuleNotFoundError: No module named 'library'

seems to be a problem for lots of people. the search path for modules is not getting set right. it should probably be happening in some global file that all the other python files load, but somehow it's not working consistently. i had this problem with BLIP and i put the following two lines in the blip python file right above the load command for 'library', as shown in this post:

https://github.com/bmaltais/kohya_ss/issues/1421#issuecomment-1693875707

you will need to put that into tag_images_by_wd14_tagger.py, and maybe if it fails in some other file, that file as well. i'm sure there is a more correct solution to this problem though, i just don't know what it is.
